CoreScriptsRoot/CoreScripts/PurchasePromptScript2.lua

```diff
@@ -399,6 +462,53 @@ PurchaseDialog.Parent = RobloxGui
 			xOffset = xOffset + 32
 		end
 
+--Gui State Control
+local function setItemDescriptionText(text)
+	if not text or type(text) ~= "string" then
+		ItemDescriptionText.Visible = false
+		ItemDescriptionText.Text = ""
+	else
+		ItemDescriptionText.Visible = true
+		ItemDescriptionText.Text = text
+	end
+end
+
+local function setPostBalanceText(text)
+	if not text or type(text) ~= "string" then
+		PostBalanceText.Visible = false
+		PostBalanceText.Text = ""
+	else
+		local y = 140
+		if CostText.Visible then
+			y = CostText.Position.Y.Offset + CostText.TextBounds.y + 6
+		end
+		PostBalanceText.Visible = true
+		PostBalanceText.Position = UDim2.new(0, 10, 0, y)
+		PostBalanceText.Text = text
+	end
+end
+
+local function setCostText(amount)
+	if not amount or type(amount) ~= "number" then
+		RobuxIcon.Visible = false
+		CostText.Visible = false
+		CostText.Text = ""
+	else
+		local textSize = TextService:GetTextSize(
+			ItemDescriptionText.Text,
+			isTenFootInterface and 48 or 18,
+			ItemDescriptionText.Font,
+			ItemDescriptionText.AbsoluteSize)
+
+		RobuxIcon.Visible = true
+		RobuxIcon.Position = UDim2.new(0, isTenFootInterface and 110*scaleFactor or 110, 0, ItemDescriptionText.Position.Y.Offset + textSize.y + (isTenFootInterface and 6*scaleFactor or 6))
+
+		CostText.Visible = true
+		CostText.Position = UDim2.new(0, isTenFootInterface and 134*scaleFactor or 134, 0, ItemDescriptionText.Position.Y.Offset + textSize.y + (isTenFootInterface and 15*scaleFactor or 15))
```

Comment:
instead of all of these ternary operators, couldn't we just always multiply by scaleFactor? I assume it is 1 unless ten foot is on.